d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -305,9 +306,11 @@ void startRpcStreamLocal()
   @Override
   public void shutdown()
   {
-    _executorService.execute(() ->
+    // create a new executor instead of using the _executorService because this shutdown should run
+    // regardless of whether _executorService is available.
+    Executors.newSingleThreadExecutor().execute(() ->
```

Comment:
actually, this new executor needs to be shutdown too. @ZoabKapoor

Reply:
Oh no! When should it be shutdown, maybe immediately after the executed command returns? 

Though from the docs it looks like `execute` is [async](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executor.html#execute-java.lang.Runnable-) and so we'd have to explicitly wait for it to return